### PR TITLE
IOFiber#toString

### LINKED ISSUE #3463: `IOFiberSpec` is still flaky in CI
```
[error]   x toString a completed fiber
[2135](https://github.com/typelevel/cats-effect/actions/runs/4284923854/jobs/7462682986#step:23:2136)
[error]    'cats.effect.IOFiber@4ad52419 RUNNING' doesn't match 'cats.effect.IOFiber@[0-9a-f][0-9a-f]+ COMPLETED' (IOFiberSpec.scala:68)
[2136](https://github.com/typelevel/cats-effect/actions/runs/4284923854/jobs/7462682986#step:23:2137)
```

We should consider disabling it in CI or something because we can't have jobs randomly failing like this.

### PR DESCRIPTION
**Pull request by durban**
Hopefully fixes #3463.

### CODE REVIEW

File: tests/shared/src/test/scala/cats/effect/IOFiberSpec.scala [hunk 1]
```diff
@@ -39,14 +41,19 @@ class IOFiberSpec extends BaseSpec with DetectPlatform {
       }
 
       "toString a suspended fiber" in real {
-        def foreverNever = IO.async_[Unit](_ => ())
+        // these are separate methods to have them in the trace:
+        def completeLatch(latch: CountDownLatch) =
+          IO.apply(latch.countDown())
+        def foreverNever(latch: CountDownLatch) =
+          IO.async[Unit](_ => completeLatch(latch).as(Some(IO.unit)))
```
Comment (armanbilge) [inline tests/shared/src/test/scala/cats/effect/IOFiberSpec.scala]: Can we use a `cats.effect.std.CountdownLatch`?

Comment (durban) [reply tests/shared/src/test/scala/cats/effect/IOFiberSpec.scala]: We could even use a `Deferred`. But it messes up the trace. But CDL clearly doesn't work (CI failure), so... I'll try `std.CountDownLatch`...

Comment (armanbilge) [reply tests/shared/src/test/scala/cats/effect/IOFiberSpec.scala]: Oh right! How does `Deferred` mess up the trace? As long as it still says `SUSPENDED` that's what matters, the current test may be too particular.

Comment (durban) [reply tests/shared/src/test/scala/cats/effect/IOFiberSpec.scala]: Yeah, the `SUSPENDED` is okay; the method name is non-deterministically incorrect. But I've removed the latch completely, it doesn't seem to actually help...

**Review by armanbilge — APPROVED** [review]
Nice, thanks so much for quashing this one!!

**Review by djspiewak — APPROVED** [review]

**Pull request merged**